### PR TITLE
Upgraded sslyze from v4.1.0 to v5.0.2

### PR DESCRIPTION
**Pull request by secureCodeBoxBot**
This is Pull Request upgrades sslyze from v4.1.0 to v5.0.2
* We now use a custom Docker image for SSLyze to work around some limitations of the software. The switch will be performed automatically when you update the scanner using Helm, you do not need to perform any manual actions.
* SSLyze changed its JSON output schema, which will be reflected in the raw output of the scanner. You can find the new output schema [here](https://github.com/nabla-c0d3/sslyze/blob/release/json_output_schema.json). The secureCodeBox findings format remains unchanged, so if you rely only on the findings, not the raw output, you will not need to make any changes based on this.
* **BREAKING :** SSLyze removed or replaced a number of parameters. In particular:
`--regular` was replaced with `--mozilla_config {modern,intermediate,old}`
`--resum_rate` was replaced with `--resum_attempts`
* New commmand-line options:
`--elliptic_curves` Test a server for supported elliptic curves.

* **BREAKING:** Public-Key-Pins headers are no longer checked by SSLyze, as the pinning feature has been removed from most browsers
* This release also includes other changes that don't modify SCB's functionality. Info can be found in the link below

**Official Changelogs:** [5.0.0](https://github.com/nabla-c0d3/sslyze/releases/tag/5.0.0), [5.0.1 (contains another parameter change)](https://github.com/nabla-c0d3/sslyze/releases/tag/5.0.1), [5.0.2](https://github.com/nabla-c0d3/sslyze/releases/tag/5.0.2)

### CODE REVIEW

**Comment by malexmave** [comment]
Okay, I assume the problem with the changelog here is that a star (`*`) is interpreted as a wildcard file operator somehow? Paging @Ilyesbdlala :).

**Comment by malexmave** [comment]
Converting this to draft. The update changed the JSON format, the reason the CI runs is probably that the unit tests are unaffected by this change and I am not sure that integration tests exist for this. We'll have to take a detailed look at this change before merging it.

**Comment by Ilyesbdlala** [comment]
> Okay, I assume the problem with the changelog here is that a star (`*`) is interpreted as a wildcard file operator somehow? Paging @Ilyesbdlala :).

It does seems so. I didn't know this was a function in the "GitHub's" Markdown. I'll look into it.
Actual release log : [here](https://github.com/nabla-c0d3/sslyze/releases/tag/5.0.0)

> Converting this to draft. The update [changed the JSON format](https://github.com/nabla-c0d3/sslyze/releases/tag/5.0.0), the reason the CI runs is probably that the unit tests are unaffected by this change and I am not sure that integration tests exist for this. We'll have to take a detailed look at this change before merging it.

By sslyze, the appVersion tag is actually overwitten by `scanner.image.tag` in the values.yaml file as I mentioned here #429.
Since this time the docker image 5.0.0 is released. We can simply make the necessary changes to the parser and upgrade sslyze directly to 5.0.0.

**Comment by Ilyesbdlala** [comment]
This PR includes Sslyze to  custom build scanners. The custom image is there to avoid non-zero exit codes.
New JSON output schema : https://github.com/nabla-c0d3/sslyze/blob/release/json_output_schema.json
The following commmand-line options are no longer available:
	`--regular` replaced with `--mozilla_config {modern,intermediate,old}`
	`--resum_rate` replaced with `--resum_attempts`
	`--version`
New commmand-line options:
	`--elliptic_curves` Test a server for supported elliptic curves.

File: scanners/sslyze/parser/parser.js [hunk 1]
```diff
@@ -28,8 +28,8 @@ function parse(fileContent) {
     ...analyseCertificateDeployments(serverScanResult),
   ];
 
-  const serverInfo = serverScanResult.server_info;
-  const { ip_address, hostname, port } = serverInfo.server_location;
+  // const serverInfo = serverScanResult.server_info;
```
Comment (malexmave) [inline scanners/sslyze/parser/parser.js:31]: Can this be deleted?

File: scanners/sslyze/parser/parser.test.js [hunk 2]
```diff
@@ -32,24 +32,31 @@ test("parses result file for www.securecodebox.io correctly", async () => {
     hint: null,
     reference: null,
     location: "www.securecodebox.io:443",
-    attributes: {
+    attributes: {
```
Comment (malexmave) [inline scanners/sslyze/parser/parser.test.js:35]: This file has a bunch of whitespaces in weird places (see rest of the diff as well).

File: scanners/sslyze/parser/parser.test.js [hunk 3]
```diff
@@ -396,10 +422,10 @@ test("parses result file for self-signed.badssl.com correctly", async () => {
     reference: null,
   });
 });
-
+/*
```
Comment (malexmave) [inline scanners/sslyze/parser/parser.test.js:425]: Commented-out test => can this be re-added or deleted?

File: scanners/sslyze/parser/parser.test.js [hunk 4]
```diff
@@ -408,7 +434,10 @@ test("parses an empty result file correctly", async () => {
   await expect(validateParser(findings)).resolves.toBeUndefined();
   expect(findings).toEqual([]);
 });
+*/
 
+// Commented out until I can reproduce the test files
+/*
```
Comment (malexmave) [inline scanners/sslyze/parser/parser.test.js:440]: Same question here: Can this be re-added or deleted?

**Review by malexmave — CHANGES_REQUESTED** [review]
Thanks for your work on this! Since you added it to the "to review" state, I took a quick look at it and have a few minor points, just to avoid having dead or strangely formatted code in the repo.

File: scanners/sslyze/parser/parser.js (hunk 1 above)
Comment (Ilyesbdlala) [reply scanners/sslyze/parser/parser.js]: Yes. I deleted it.

File: scanners/sslyze/parser/parser.test.js (hunk 2 above)
Comment (Ilyesbdlala) [reply scanners/sslyze/parser/parser.test.js]: My bad, I had to manually format a few parts. it should be fine now

File: scanners/sslyze/parser/parser.test.js (hunk 3 above)
Comment (Ilyesbdlala) [reply scanners/sslyze/parser/parser.test.js]: Yes. I just readded it

File: scanners/sslyze/parser/parser.test.js (hunk 4 above)
Comment (Ilyesbdlala) [reply scanners/sslyze/parser/parser.test.js:438]: I don't know how to reproduce the mixed-connectivity test file. So I can't readd it just yet. Can you help with that ?

Comment (malexmave) [reply scanners/sslyze/parser/parser.test.js:438]: Not sure about this, either. Maybe @J12934 knows more once he's back.

Comment (malexmave) [reply scanners/sslyze/parser/parser.test.js:438]: I think after our discussion with @J12934, you can just remove this test case and associated files for the moment.

Comment (Ilyesbdlala) [reply scanners/sslyze/parser/parser.test.js]: I don't know why I thought i already did this, but it's done now ^^ .

**Comment by malexmave** [comment]
Does this PR change anything about the external interface of this scanner from the point of view of an SCB user? I.e.,

- How it is parameterized (in a non-backwards-compatible way)
- Remove or change the meaning of part of the finding format
- ... basically anything that would require users to take manual action when updating to this version?

I *think* it does not, but you are the expert, @Ilyesbdlala.

**Review by malexmave — APPROVED** [review]

**Comment by malexmave** [comment]
Also, can anyone see the findings that Snyk complains about for this PR? I don't have an account there.

**Comment by J12934** [comment]
@malexmave  seems like its being rate limited, i'd be ok ignoring these errors

Screenshot 2022-01-12 at 13 15 44

**Review by Ilyesbdlala — APPROVED** [review]
Reapproved the PR. That is since no changes have been made since @malexmave review.

**Pull request merged**